A printf format specifier of "qx" isn't supported on Solaris.

http://man7.org/linux/man-pages/man3/printf.3.html says:

  q   ("quad". 4.4BSD and Linux libc5 only.  Don't use.)  This is a
      synonym for ll.

This change will be sent upstream.

--- util-linux-2.25.2/text-utils/hexdump-parse.c.orig	2015-02-16 18:06:45.921790582 -0800
+++ util-linux-2.25.2/text-utils/hexdump-parse.c	2015-02-16 18:09:59.879541819 -0800
@@ -226,7 +226,7 @@
 	struct hexdump_fu *fu;
 	struct list_head *p, *q;
 	char *p1, *p2, *fmtp;
-	char savech, cs[3];
+	char savech, cs[4];
 	int nconv, prec = 0;
 
 	list_for_each (p, &fs->fulist) {
@@ -301,9 +301,10 @@
 				goto isint;
 			} else if (first_letter(cs, "ouxX")) {
 				pr->flags = F_UINT;
-isint:				cs[2] = '\0';
-				cs[1] = cs[0];
-				cs[0] = 'q';
+isint:				cs[3] = '\0';
+				cs[2] = cs[0];
+				cs[1] = 'l';
+				cs[0] = 'l';
 				switch(fu->bcnt) {
 					case 0:
 						pr->bcnt = 4;
@@ -355,9 +356,10 @@
 						pr->flags = F_ADDRESS;
 						++p2;
 						if (first_letter(p1 + 2, "dox")) {
-							cs[0] = 'q';
-							cs[1] = p1[2];
-							cs[2] = '\0';
+							cs[0] = 'l';
+							cs[1] = 'l';
+							cs[2] = p1[2];
+							cs[3] = '\0';
 						} else {
 							p1[3] = '\0';
 							badconv(p1);
